### DISCUSSION TITLE
SVG not rendering in Jupyter notebooks (ipynb)

### DISCUSSION THREAD

**Discussion by u1735067** [community/community#12355] [Other Feature Feedback, Questions, & Ideas] [answered]
Hi,

a bit like https://github.com/github/feedback/discussions/5546 (cc @el-mapache maybe ?), in Jupyter Notebook files (.ipynb), it seems SVG images are not rendered anymore, both on Github and Github Gists.

An example is: https://github.com/ipython/ipython/blob/master/examples/IPython%20Kernel/Rich%20Output.ipynb (see `Out[8]`)

Another example, on Gists this time: ~~https://github.com/HuifangYeo/test_svg/blob/a67132b409e8889551188c7c831bf67d9555de08/svg.ipynb~~ https://gist.github.com/greglandrum/d85d5693e57c306e30057ec4d4d11342

This issue was also reported on the Community forum at https://github.community/t/inline-svg-in-jupyter-notebook-no-longer-get-displayed/219156/3

Could you bring back this feature please? 😊

**Comment by el-mapache** [comment, marked as answer]
Hi u1735067!

Thanks for reaching out. I'm sorry to hear the feature isn't working as expected for you!

The issue you linked to is actually handled by a different service, and probably unrelated. However, I would expect inline SVGs to render properly within a python notebook.

We'll create a tracking issue and start triaging this. Someone from our team will be sure to keep you updated as we make progress.

Thanks!

**Reply by u1735067** [reply]
Thanks to you :) (and sorry for the unrelated mention then)

I realize I mistaken the second example as a Gist example, so here is an actual ipynb Gist failing to display SVG: https://gist.github.com/greglandrum/d85d5693e57c306e30057ec4d4d11342 (Out 10 and 11)

**Reply by u1735067** [reply]
Hi, I don't know if somebody already looked at this issue, but while it seems better, I still have some issue displaying graphs

**Reply by el-mapache** [reply]
@u1735067 Apologies for the delay in getting back to you! There are still certain svg tags being removed, and we'll look into it shortly. Thanks!

**Reply by el-mapache** [reply]
@u1735067 Thanks again for your feedback! I've checked back in our the gist you linked, and I believe we've resolved the chart rendering. Please feel free to reach out again if you notice other output rendering in unexpected ways.

**Reply by u1735067** [reply]
Thanks to you, it seems to be rendering correctly :)

**Reply by manuelinfosec** [reply]
It ain't working, fam.

**Reply by u1735067** [reply]
You're right, it's broken again :(
@el-mapache maybe?

**Reply by manuelinfosec** [reply]
I thought it was a problem from my end. I had to make a lot of changes to my project now, to no avail. @github-staff, please look into this. Thanks.

**Reply by el-mapache** [reply]
There is a regression in this behavior, we'll be triaging it. Thanks for bringing it to our attention!

**Reply by therzka** [reply]
👋 I've just pushed a change that should resolve the issue with SVGs. Let us know if anything else crops up!

**Reply by u1735067** [reply]
@therzka : Thank you, it's not quite fully fixed however, on one of my gists I'm seeing errors like this in the console:
```
Error: <path> attribute d: Expected path command, "…0125 311.010244 \\r\\nL 715.6125 3…".
sanitizeNotebook @ html-sanitizer.ts:142
setupIframeResizeMessenger @ iframe-messenger.ts:20
(anonymous) @ index.js:5
(anonymous) @ bundle-dc8ceacd1312012068f2.js:2791
(anonymous) @ bundle-dc8ceacd1312012068f2.js:2793
```

**Reply by therzka** [reply]
> @therzka : Thank you, it's not quite fully fixed however, on one of my gists I'm seeing errors like this in the console:
> 
> ```
> Error: <path> attribute d: Expected path command, "…0125 311.010244 \\r\\nL 715.6125 3…".
> sanitizeNotebook @ html-sanitizer.ts:142
> setupIframeResizeMessenger @ iframe-messenger.ts:20
> (anonymous) @ index.js:5
> (anonymous) @ bundle-dc8ceacd1312012068f2.js:2791
> (anonymous) @ bundle-dc8ceacd1312012068f2.js:2793
> ```

could you link me to the gist where you're seeing this?

**Reply by u1735067** [reply]
> could you link me to the gist where you're seeing this?

Yep, it's about the last graph of https://gist.github.com/u1735067/ce590a1d7d4f64e028a40d7520ce3964

**Reply by el-mapache** [reply]
@u1735067 Thanks for your patience with this issue. We've rolled out some changes which should make the majority of svg charts render correctly again. 

We are still seeing issue with SVGs generated with `gnuplot`. This seems to stem from extra escaping within nbconvert. Hoping we can address this soon.

**Reply by XuehaiPan** [reply]
Hi, the texts in SVGs are not well rendered in IPython Notebooks.

For example:

https://github.com/metaopt/torchopt/blob/main/tutorials/2_Visualization.ipynb

<img width="321" alt="image" src="https://user-images.githubusercontent.com/16078332/207584607-61fa65c8-d2d9-4b84-8fed-2a1672d49a23.png">

The same notebook on Colab:

https://colab.research.google.com/github/metaopt/torchopt/blob/main/tutorials/2_Visualization.ipynb

<img width="451" alt="image" src="https://user-images.githubusercontent.com/16078332/207584716-23e55eb4-c863-433c-8fc7-9cc08e64e168.png">

**Reply by el-mapache** [reply]
@XuehaiPan Thanks for the report! Text should now be rendering more or less correctly within svgs. Note that if your text includes things that resemble HTML tags e.g. `<a>` or `<c>`, our sanitizer will strip those and they will not be rendered in the final output.

We are actively working on strategies to improve the rendering fidelity of notebooks, thank you for being a part of the discussion!

**Reply by cben** [reply]
If you use nbconvert and/or nbviewer under the hood, FYI nbviewer [fixed](https://github.com/jupyter/nbviewer/issues/1037) SVG rendering by [upgrading to newer nbconvert](https://github.com/jupyter/nbviewer/pull/1079).

E.g. https://nbviewer.org/github/cben/sandbox/blob/master/jupyter_notebook/mermaid_mime_type.ipynb  
vs GitHub's https://github.com/cben/sandbox/blob/master/jupyter_notebook/mermaid_mime_type.ipynb

Note that works by `<img src="data:image/svg+xml;base64,...` which was [mentioned here](https://github.com/jupyter/nbconvert/pull/2018#issuecomment-1931024762) as a problem for GitHub.
